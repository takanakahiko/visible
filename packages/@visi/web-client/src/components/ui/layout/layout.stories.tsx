import React from 'react';

import { Layout } from '.';

export default {
  title: 'Layout',
  component: Layout,
};

export const Normal = () => (
  <div>
    <Layout.Header>header</Layout.Header>

    <Layout.Content>
      <Layout.Main>main</Layout.Main>
      <Layout.Aside>aside</Layout.Aside>
    </Layout.Content>

    <Layout.Footer>footer</Layout.Footer>
  </div>
);
