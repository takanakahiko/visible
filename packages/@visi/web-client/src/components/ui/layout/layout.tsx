import classNames from 'classnames';
import React from 'react';

export type HeaderProps = JSX.IntrinsicElements['header'];

const Header = ({ children, className, ...rest }: HeaderProps) => {
  return (
    <header
      className={classNames('bg-primary-500', 'shadow', 'z-10', className)}
      {...rest}
    >
      {children}
    </header>
  );
};

export type FooterProps = JSX.IntrinsicElements['footer'];

const Footer = ({ children, className, ...rest }: FooterProps) => {
  return (
    <footer
      className={classNames(
        'border-t',
        'border-gray-300',
        'bg-gray-200',
        className,
      )}
      {...rest}
    >
      {children}
    </footer>
  );
};

export type MainProps = JSX.IntrinsicElements['main'];

const Main = ({ children, className, ...rest }: MainProps) => {
  return (
    <main className={classNames('w-full', className)} {...rest}>
      {children}
    </main>
  );
};

export type ContainerProps = JSX.IntrinsicElements['div'];

const Container = ({ children, className, ...rest }: ContainerProps) => {
  return (
    <div
      className={classNames(
        'flex',
        'flex-wrap',
        'w-full',
        'mx-auto',
        'box-border',
        'p-3',
        'md:max-w-screen-md',
        'lg:max-w-screen-lg',
        'xl:max-w-screen-xl',
        className,
      )}
      {...rest}
    >
      {children}
    </div>
  );
};

// Container.defaultProps = {
//   padding: true,
// };

export type ContentProps = JSX.IntrinsicElements['div'];

const Content = ({ children, className, ...rest }: ContentProps) => {
  return (
    <article className={classNames('flex-1', className)} {...rest}>
      {children}
    </article>
  );
};

export type AsideProps = JSX.IntrinsicElements['aside'];

const Aside = ({ children, className, ...rest }: AsideProps) => {
  return (
    <aside
      className={classNames(
        'w-full',
        'lg:ml-8',
        'lg:w-3/12',
        'space-y-8',
        className,
      )}
      {...rest}
    >
      {children}
    </aside>
  );
};

export const Layout = {
  Header,
  Main,
  Container,
  Content,
  Aside,
  Footer,
};
