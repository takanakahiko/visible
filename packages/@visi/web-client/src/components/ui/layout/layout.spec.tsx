import { render } from '@testing-library/react';
import React from 'react';

import { Normal } from './layout.stories';

describe('Layout', () => {
  it('matches snapshot', () => {
    const { container } = render(<Normal />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
