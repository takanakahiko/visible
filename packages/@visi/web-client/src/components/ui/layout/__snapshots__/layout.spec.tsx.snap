// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<div>
  <header
    class="bg-primary-500 shadow z-10"
  >
    header
  </header>
  <article
    class="flex-1"
  >
    <main
      class="w-full"
    >
      main
    </main>
    <aside
      class="w-full lg:ml-8 lg:w-3/12 space-y-8"
    >
      aside
    </aside>
  </article>
  <footer
    class="border-t border-gray-300 bg-gray-200"
  >
    footer
  </footer>
</div>
`;
